src/main/java/sirius/biz/translations/BasicTranslations.java

```diff
@@ -0,0 +1,223 @@
+/*
+ * Made with all the love in the world
+ * by scireum in Remshalden, Germany
+ *
+ * Copyright by scireum GmbH
+ * http://www.scireum.de - info@scireum.de
+ */
+
+package sirius.biz.translations;
+
+import sirius.biz.protocol.JournalData;
+import sirius.biz.protocol.Journaled;
+import sirius.db.mixing.BaseEntity;
+import sirius.db.mixing.Composite;
+import sirius.db.mixing.FieldLookupCache;
+import sirius.db.mixing.Mapping;
+import sirius.db.mixing.annotations.AfterDelete;
+import sirius.db.mixing.annotations.Transient;
+import sirius.kernel.commons.Strings;
+import sirius.kernel.di.std.ConfigValue;
+import sirius.kernel.di.std.Part;
+
+import javax.annotation.Nonnull;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * Provides database independent specifications for functionality to access and edit translations for an entity.
+ *
+ * @param <T> the translation entity type used by a concrete subclass
+ */
+public abstract class BasicTranslations<T extends BaseEntity<?> & Translation> extends Composite {
+
+    @Transient
+    protected BaseEntity<?> owner;
+
+    @Transient
+    @ConfigValue("mixing.multiLanguageStrings.supportedLanguages")
+    protected static Set<String> supportedLanguages;
+
+    @Transient
+    @Part
+    private static FieldLookupCache fieldLookupCache;
+
+    protected BasicTranslations(BaseEntity<?> owner) {
+        this.owner = owner;
+    }
+
+    /**
+     * Ensures that translations are cascaded-deleted upon deletion of the owner entity.
+     */
+    @AfterDelete
+    protected abstract void removeTranslations();
+
+    /**
+     * Provides a validator so translations can only be added for supported languages.
+     *
+     * @param lang the language code in question
+     * @return true if language is supported by the system, false otherwise
+     */
+    protected boolean isSupportedLanguage(String lang) {
+        return supportedLanguages.contains(lang);
+    }
+
+    /**
+     * Updates the translation for the given field and language, or adds a new translation if it does not exist.
+     * <p>
+     * Note that an existing translation is deleted when calling this with empty text.
+     * <p>
+     * Also note that this will emit an {@link sirius.biz.protocol.JournalEntry} to
+     * {@link sirius.biz.protocol.Journaled} owners keeping track of the changes made to their translations.
+     *
+     * @param field {@link Mapping} of the translated field
+     * @param lang  language code
+     * @param text  translated text for the given language
+     */
+    public void updateText(@Nonnull Mapping field, String lang, String text) {
+        if (Strings.isEmpty(text)) {
+            deleteText(field, lang);
+            return;
```

Comment:
The question is if deleting should only be allowed via updateText() with empty text, or rather via explicit calls to deleteText(). Currently, both options are supported so writing to the journal in deleteText() makes more sense.
It could be moved, if only one of the options should be publicly available